services/ui-src/src/components/reports/ReportPage.tsx

```diff
@@ -76,6 +78,20 @@ export const ReportPage = ({ route }: Props) => {
     }
   };
 
+  const getEntities = (formEntityType: PageEntityType) => {
+    const entityTypeMap = {
+      plan: "aap-1", // reportData fieldId of the entity array
+      bssEntity: "absse-a8",
```

Comment:
This is exactly what I was envisioning. We'll need to modify because the field names will be changing to `plans` and `bssEntities`, so I wonder if the form JSON prop maybe becomes like `mappedEntityField` or something.